nomad/structs/config/tls.go

```diff
@@ -72,6 +77,14 @@ func (k *KeyLoader) GetOutgoingCertificate(*tls.ClientHelloInfo) (*tls.Certifica
 	return k.Certificate, nil
 }
 
+func (t *TLSConfig) GetKeyLoader() *KeyLoader {
+	// If the keyloader has not yet been initialized, do it here
+	if t.KeyLoader == nil {
+		t.KeyLoader = &KeyLoader{}
```

Comment:
This cleans up so much! Probably should lock this too

Reply:
Ah this would have to be outside the if statement. It is ensuring that if you have two concurrent calls where the keyloader isn't made they both return the same object.

Reply:
Ah, of course. Will fix that up.